client/components/domains/register-domain-step/index.jsx

```diff
@@ -584,20 +583,36 @@ const RegisterDomainStep = React.createClass( {
 				}
 				break;
 
-			case 'mappable_but_forbidden_subdomain':
+			case 'not_mappable_forbidden_subdomain':
 				message = this.translate( 'Subdomains starting with \'www.\' cannot be mapped to a WordPress.com blog' );
 				break;
 
-			case 'mappable_but_mapped_domain':
+			case 'not_mappable_invalid_tld':
+			case 'invalid_query':
+				message = this.translate( 'Sorry, %(domain)s does not appear to be a valid domain name.', {
+					args: { domain: domain }
+				} );
+				break;
+
+			case 'not_mappable_mapped_domain':
 				message = this.translate( 'This domain is already mapped to a WordPress.com site.' );
 				break;
 
-			case 'mappable_but_restricted_domain':
+			case 'not_mappable_restricted_domain':
```

Comment:
There is a discrepancy between the error code and the message we are showing to the user. If it's guaranteed that "restricted" domain means WordPress.com subdomain, we should change the error code as such, and if not, we should change the error message as such.